Tasks/GoTool/gotool.ts

```diff
@@ -0,0 +1,131 @@
+///<reference path="../../definitions/node.d.ts" />
+
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as tl from 'vsts-task-lib/task';
+import * as os from 'os';
+import * as path from 'path';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+try {
+    tl.setResourcePath(path.join(__dirname, "task.json"));
+} catch (error) {
+    tl.setResult(tl.TaskResult.Failed, error);
+}
+
+async function run() {
+    try {
+        let versionSpec = tl.getInput('version', true);
+        await getGo(versionSpec);
+    }
+    catch (error) {
+        tl.setResult(tl.TaskResult.Failed, error);
+    }
+}
+
+async function getGo(version: string) {
+    // check cache
+    let toolPath: string;
+    toolPath = toolLib.findLocalTool('go', version);
+
+    if (!toolPath) {
+        // download, extract, cache
+        toolPath = await acquireGo(version);
+    }
+
+    //
+    // a tool installer initimately knows details about the layout of that tool
+    // for example, node binary is in the bin folder after the extract on Mac/Linux.
+    // layouts could change by version, by platform etc... but that's the tool installers job
+    //
+    if (osPlat != 'win32') {
+        toolPath = path.join(toolPath, 'bin');
+    }
+
+    //
+    // prepend the tools path. instructs the agent to prepend for future tasks
+    //
+    toolLib.prependPath(toolPath);
+    setGoEnvironmentVariables(toolPath);
```

Comment:
To make it more clear, you can modify the function name to setGoProcessEnvironmentVariables, since here you are setting variables for the process only.